src/Helpers/auth_helper.php

```diff
@@ -43,3 +44,33 @@ function user_id()
         return $auth->id();
     }
 }
+
+if (! function_exists('shieldSetting')) {
+    /**
+     * Provides a wrapper function for settings module.
+     *
+     * @param mixed $value
+     *
+     * @return array|bool|float|int|object|string|void|null
+     * @phpstan-return ($value is null ? array|bool|float|int|object|string|null : void)
+     */
+    function shieldSetting(?string $key = null, $value = null)
+    {
+        /** @var AuthConfig $config */
+        $config = config('Auth');
+        if($config->useSettings) {
+            return setting($key, $value);
+        }
+
+        // Getting the value?
+        if (!empty($key) && count(func_get_args()) === 1) {
+            $parts = explode('.', $key);
+            if (count($parts) === 1) {
+                throw new InvalidArgumentException('$key must contain both the class and field name, i.e. Foo.bar');
+            }
+            return config($parts[0])?->{$parts[1]} ?? null;
+        }
+
+        throw new InvalidArgumentException('Settings library is not being used for shield');
```

Comment:
This capturing exception feels off. This would be reached if:

1. `Auth::$useSettings` is true but `settings` is not installed.
2. `$useSettings` is false but `$key` is null or empty string.
3. `$useSettings` is false, `$key` is non-empty-string, and $value is provided

These different scenarios should be tackled separately within each conditional above.